content/learn/book/your-first-game/_index.md

```diff
@@ -0,0 +1,685 @@
+
+Finally execute this in your project folder:
+
+```bash
+cargo run
+```
+You will probably get 3 warnings. They are okay for now, we will have them fixed by the end of this tutorial.  
+
+You should be able to move the paddle with your cursor keys and it should look like this:
+
+![screenshot2](breakout2.png)
+
+Okay, so what is the code doing?
+
+```rs
+use bevy::{
+    prelude::*,
+    render::pass::ClearColor,
+};
+```
+This simply declares a couple of local name bindings to the most common Bevy types and plugins as well as a local name binding to ClearColor.
+ClearColor is a struct that contains the color we will use to clear the game with; effectively becoming the background color of our game.
+
+```rs
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .insert_resource(ClearColor(Color::rgb(0.9, 0.9, 0.9)))
+        .add_startup_system(setup.system())
+        .add_system(paddle_movement_system.system())
+        .run();
+}
+```
+This is the main entry point into our program. This function will be called once the program is successfully compiled and executed.
+
+[App::build()](https://github.com/bevyengine/bevy/blob/main/crates/bevy_app/src/app.rs) is a very simple function that just returns a new instance of [AppBuilder](https://docs.rs/bevy/0.4.0/bevy/app/struct.AppBuilder.html)
+
+As the other function calls suggest, the AppBuilder allows us to add plugins and (startup) systems and to insert resources.
+Finally a call to run starts the game engine with all of the configured systems, plugins and resources.
+The [getting started guide](/learn/book/getting-started/) covers these types in great detail.
+
+```rs
+add_plugins(DefaultPlugins)
+``` 
+This call simply adds all the default plugins to the app.  
+Follow this link if you are interested in a complete list of all of the default plugins: [default_plugins.rs](https://github.com/bevyengine/bevy/blob/latest/crates/bevy_internal/src/default_plugins.rs)
+
+```rs
+.insert_resource(ClearColor(Color::rgb(0.9, 0.9, 0.9)))
+``` 
+Here we insert a ClearColor resource, set to a light grey. (rgb stands for the red, green and blue parts of the final color.)
+
+```rs
+.add_startup_system(setup.system())
+``` 
+Here we register our very own setup function as a startup system.
+Startup systems are only called once and before any of the normal systems are called.
+We will cover the setup function code in detail in a moment.
+
+```rs
+.add_system(paddle_movement_system.system())
+``` 
+Here we add our very own paddle_movement_system function as a system.
+Systems are called every frame.
+We will cover the paddle_movement_system function code in detail in a moment.
+
+```rs
+.run();
+``` 
+Finally we tell the Bevy game engine to start with a call to run.  
+(If you are really interested in the source code of the run function you can find it here: [app.rs](https://github.com/bevyengine/bevy/blob/latest/crates/bevy_app/src/app.rs))
```

Comment:
```suggestion
(If you're curious about the details of how the `run` function works its magic, you can follow the source code trail that begins in [app.rs](https://github.com/bevyengine/bevy/blob/latest/crates/bevy_app/src/app.rs))
```

Reply:
All your changes are great, but this one is especially awesome. I love your word choices! Much improved!